src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -170,7 +170,13 @@ static SingleValue GetTypeValueNodeFromGenericArgument (SyntaxNodeAnalysisContex
 				SymbolKind.TypeParameter => new GenericParameterValue ((ITypeParameterSymbol) type),
 				// Technically this should be a new value node type as it's not a System.Type instance representation, but just the generic parameter
 				// That said we only use it to perform the dynamically accessed members checks and for that purpose treating it as System.Type is perfectly valid.
-				SymbolKind.NamedType => new SystemTypeValue (new TypeProxy ((INamedTypeSymbol) type)),
+				SymbolKind.NamedType 
+					=> ((INamedTypeSymbol)type).NullableAnnotation switch {
```

Comment:
I find this pretty weird - so roslyn doesn't translate nullable value types to true `Nullable<T>` type symbols?
If not, then this will need to somehow differentiate ref types and not do this for those.

Reply:
It looks like that `NullableAnnotation` isn't reliable for our purposes, so I had to change it to check the name and namespace instead.